src/Native/Json.js

```diff
@@ -137,9 +137,11 @@ Elm.Native.Json.make = function(localRuntime) {
 			if (isObject) {
 				var keyValuePairs = List.Nil;
 				for (var key in value) {
-					var elmValue = decoder(value[key]);
-					var pair = Utils.Tuple2(key, elmValue);
-					keyValuePairs = List.Cons(pair, keyValuePairs);
+					if (value.hasOwnProperty(key)) {
```

Comment:
Sure, but decoders won't let functions through, so that case never happens. So that's covered by (a)